core/src/main/java/hudson/model/LoadBalancer.java

```diff
@@ -112,7 +112,14 @@ public String hash(ExecutorChunk node) {
         private boolean assignGreedily(Mapping m, Task task, List<ConsistentHash<ExecutorChunk>> hashes, int i) {
             if (i==hashes.size())   return true;    // fully assigned
 
-            String key = task.getAffinityKey() + (i>0 ? String.valueOf(i) : "");
+            String key;
+            try {
+                key = task.getAffinityKey();
+            } catch (RuntimeException e) {
```

Comment:
```suggestion
            } catch (RuntimeException e) {
                LOGGER.log(Level.WARNING, null, e);
```

(plus introduction of that field)

Reply:
I considered this, but `Queue.maintain` runs every 5 seconds, so if the error persists this starts to seriously spam logs, and I really hate looking through support bundles where the logs only cover the last 30 minutes because of a single exception. Maybe we could log it at FINE, or try to come up with something to only log once for a given build?

Reply:
`FINE` for now, and bonus points for patching the standard log handlers (`winstone`, core, `support-core`) to deduplicate automatically.